openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java

```diff
@@ -276,12 +283,18 @@ private static RecordFieldWithDefaultValueNode getRecordFieldWithDefaultValueNod
      * @param fieldSchema Schema for data type
      * @return {@link MetadataNode}
      */
-    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema) {
+    public static AnnotationNode generateConstraintNode(String typeName, Schema<?> fieldSchema)
+            throws BallerinaOpenApiException {
         if (isConstraintAllowed(typeName, fieldSchema)) {
-            if (GeneratorUtils.isStringSchema(fieldSchema)) {
+            String ballerinaType = convertOpenAPITypeToBallerina(fieldSchema);
+            // For openAPI field schemas having 'string' type, constraints generation will be skipped when
+            // the counterpart Ballerina type is non-string (e.g. for string schemas with format 'binary' or 'byte',
+            // the counterpart ballerina type is 'byte[]', hence any string constraints cannot be applied)
+            if (ballerinaType.equals(STRING)) {
                 // Attributes : maxLength, minLength
                 return generateStringConstraint(fieldSchema);
-            } else if (GeneratorUtils.isNumberSchema(fieldSchema) || GeneratorUtils.isIntegerSchema(fieldSchema)) {
+            } else if (ballerinaType.equals(DECIMAL) || ballerinaType.equals(FLOAT) || ballerinaType.equals(INT) ||
+                    ballerinaType.equals(INT_SIGNED32)) {
```

Comment:
What about signed16 and signed8 integers? :)

Reply:
Swagger only offers standard [integer formats](https://swagger.io/docs/specification/data-models/data-types/#numbers:~:text=with%20double%20precision.-,integer,-%E2%80%93) in signed 32 and signed 64. If the user provides a format that is not one of these options in OpenAPI, it redirects to its main data type, 'integer' (ballerina `int`). In Ballerina, the **int** type represents signed 64, while if it's signed 32, we map it to `int:Singed32`.